app/src/main/java/com/android/sample/MainActivity.kt

```
@@ -3,41 +3,116 @@ package com.android.sample
+                // évite les doublons de back stack
+                nav.navigate(goTo) {
+                  popUpTo(nav.graph.startDestinationId) { inclusive = true }
+                  launchSingleTop = true
+                }
+              }
+          auth.addAuthStateListener(l)
+          onDispose { auth.removeAuthStateListener(l) }
+        }
+
+        Scaffold(
+            topBar = {
+              CenterAlignedTopAppBar(
+                  title = { Text(if (user == null) "EduMon — Connexion" else "EduMon — Stats") },
+                  actions = {
+                    if (user != null) {
+                      TextButton(onClick = { signOutAll() }) { Text("Déconnexion") }
+                    }
+                  })
+            }) { padding ->
+              Box(Modifier.fillMaxSize().padding(padding)) {
+                NavHost(
+                    navController = nav,
+                    startDestination = if (user == null) "login" else "stats") {
+                      composable("login") {
+                        // Écran login — quand Google/Firebase connecte, l’AuthStateListener push
+                        // vers "stats"
+                        LoginScreen()
+                      }
+
+                      composable("stats") {
+                        // VM créée uniquement sur l’écran Stats (évite NPE quand pas loggé)
+                        val statsVm: StatsViewModel = viewModel()
+
+                        // Brancher Firestore en toute sécurité (après login)
+                        LaunchedEffect(user?.uid) {
+                          user?.let {
+                            try {
+                              val repo = FirestoreStatsRepository()
+                              repo.ensureDefaults()
+                              statsVm.attachFirestore(repo.stats)
+                            } catch (_: Exception) {
+                              // en cas d’erreur Firestore: l’UI reste sur le mode Scénarios (fake)
```

Silent error handling can make debugging harder.
Add at least a Log.w("Firestore", e) or expose an error state to display in the UI.